lib/libzfs/libzfs_dataset.c

```diff
@@ -2979,6 +2980,11 @@ zfs_prop_get_int(zfs_handle_t *zhp, zfs_prop_t prop)
 	char *source;
 	uint64_t val = 0;
 
+	if ((prop == ZFS_PROP_CREATETXG) &&
+	    ((val = zfs_get_createtxg(zhp)) > 0)) {
```

Comment:
Handling this in `get_numeric_property` seems more fitting. We can also completely avoid changing the libzfs ABI by inlining `zfs_get_createtxg` here, since it isn't used anywhere else now.